examples/asset/loading_screen.rs

```diff
@@ -0,0 +1,234 @@
+use bevy::{gltf::Gltf, prelude::*};
+
+/// This example illustrates, how a loading screen can be implemented. It has an animated spinner and listens for an `AssetEvent`.
+
+#[derive(Clone, Eq, PartialEq, Debug, Hash)]
+enum GameState {
+    Loading,
+    Playing,
+}
+
+fn main() {
+    App::new()
+        .insert_resource(Msaa { samples: 4 })
+        .add_plugins(DefaultPlugins)
+        // Loading screen plugin and its `asset_listening_system`
+        .add_plugin(plugin::SimpleLoadingScreenPlugin {
+            loading_state: GameState::Loading,
+        })
+        .add_system_set(
+            SystemSet::on_update(GameState::Loading).with_system(asset_listening_system),
+        )
+        // system that sets up everything for our game
+        .add_startup_system(setup_game)
+        // system that will be run after everything is set up
+        .add_system_set(SystemSet::on_enter(GameState::Playing).with_system(game_system))
+        .run();
+}
+
+// System that listens for our `AssetEvent` and changes our game state after we finished loading our asset
+fn asset_listening_system(
+    // if you have different asset, either use another `EventReader` for your particular type of asset,
+    // or use `AssetServer::get_load_state` with a resource in which you track the handles of your assets.
+    mut asset_events: EventReader<AssetEvent<Gltf>>,
+    mut state: ResMut<State<GameState>>,
+) {
+    for event in asset_events.iter() {
+        match event {
+            AssetEvent::Created { handle: _ } => {
+                info!("Done loading, switch state to `GameState::Playing`");
+                // in this case, we know that we only had one asset (FlightHelmet.gltf#Scene0) to load,
+                // so we can switch our game state from `GameState::Loading` over to `GameState::Playing`
+                let _ = state.overwrite_set(GameState::Playing);
+            }
+            // we don't care about these events in our example
+            AssetEvent::Modified { handle: _ } => {}
+            AssetEvent::Removed { handle: _ } => {}
+        }
+    }
+}
+
+// Here we prepare our gaming scene, meaning we setup our ingame camera and load our asset.
+fn setup_game(
+    mut commands: Commands,
+    asset_server: Res<AssetServer>,
+    loading_screen: Query<&Children, With<plugin::LoadingScreen>>,
+    mut text_query: Query<&mut Text, With<plugin::LoadingScreenContent>>,
+) {
+    // Change our default loading screen text to something more descriptive
+    let children = loading_screen.single();
```

Comment:
Huh...moving everything into a `Plugin` broke this part
```
NoEntities("bevy_ecs::system::query::Query<&bevy_transform::components::children::Children, bevy_ecs::query::filter::With<loading_screen::plugin::LoadingScreen>>")'
```
Is this a system ordering problem? I guess i will just remove this here and people have to do with the default text? 😅

Reply:
Yeah, for a simple example I think it's fine to just have some default "Loading..." text. The code should be as simple as possible.